### PR TITLE
[GH-1156] - Implemented Color.ToHsv;

### PR DESCRIPTION
**Pull request by pictos**
### Description of Change ###

Added the extension method `ColorExtensions.ToHSV(this Color color, out double h, out double s, out double v);`

reference is [here](https://www.rapidtables.com/convert/color/rgb-to-hsv.html)

### Bugs Fixed ###

- Related to issue #1156 

### API Changes ###

Added: 
 
- `void ColorExtensions.ToHSV(this Color color, out double h, out double s, out double v);`

### Behavioral Changes ###

The user can now obtain the HSV values ​​for the `Color` structure.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Updated documentation ([see walkthrough](https://github.com/xamarin/Essentials/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by Redth** [comment]
@SaqibS can you give a quick look at this implementation for us?

**Comment by jamesmontemagno** [comment]
So a few things:

* Let's to `ToHsv`
* Let's copy/pasta: https://github.com/xamarin/Xamarin.Forms/blob/a715c5d06cdc0785236d503e1016cae6c6c6424e/Xamarin.Forms.Core/Color.cs#L424 :)

**Comment by pictos** [comment]
@jamesmontemagno Just to clarify, when you say that you want a method named `ToHsva` and copy/paste that code from XF, you want to convert `RGBa` to `HSVa` **or** `HSVa` to `RGBa`?

I'm asking this because I didn't find anything about converting RGBa to HSVa.

**Comment by jamesmontemagno** [comment]
I think basically grab this PR: https://github.com/xamarin/Xamarin.Forms/pull/10054/files to put in there too. We can have To and From....

File: Xamarin.Essentials/Types/PlatformExtensions/ColorExtensions.shared.cs [hunk 1]
```diff
@@ -66,6 +67,124 @@ public static Color GetComplementary(this Color color)
             // multiply by 100 as `ConvertToHsl` specifies them as values between 0 and 1.
             return ColorConverters.FromHsl(h, s * 100, l * 100);
         }
+
+        public static void ToHsv(this Color color, out double h, out double s, out double v)
```
Comment (jamesmontemagno) [inline Xamarin.Essentials/Types/PlatformExtensions/ColorExtensions.shared.cs:71]: I wonder if we can just return a Tuple instead of 3 outs?

Comment (jamesmontemagno) [reply Xamarin.Essentials/Types/PlatformExtensions/ColorExtensions.shared.cs:71]: Maybe something like:

public static (double h, double s, double v) ToHsv(Color color)

Comment (jamesmontemagno) [reply Xamarin.Essentials/Types/PlatformExtensions/ColorExtensions.shared.cs:71]: Besides that we just need docs and good to go :)

**Comment by pictos** [comment]
@jamesmontemagno it's done 🚀.

**Comment by jamesmontemagno** [comment]
WHOOP! TUPLES! Looks good! 

Do you want me to go through the docs or are you able to handle that? I know it is a bit tricky with weird tooling.

**Comment by pictos** [comment]
@jamesmontemagno I've already added the docs in this PR. Did I miss something?

**Comment by jamesmontemagno** [comment]
Seems to be mad about something... let me see.

**Comment by jamesmontemagno** [comment]
Ah, needs to be added to the indexes. See: https://github.com/xamarin/Essentials/pull/1200/files

**Comment by jamesmontemagno** [comment]
If you call "generate-docs.cmd" and it will do it.

**Comment by pictos** [comment]
@jamesmontemagno  how about now?

**Comment by jamesmontemagno** [comment]
That looks much better! will wait for it to go through :)

**Comment by pictos** [comment]
It's a pleasure ❤️

**Pull request merged**